.github/workflows/test.yml

```diff
@@ -47,12 +37,14 @@ jobs:
         pip install wheel
         pip install mypy types-requests types-PyYAML
         pip install .[tests]
+    - name: Remove build artifacts
+      run: rm -rf build dist
     - name: Start services with Docker Compose
-      run: docker compose -f docker-compose.testing.yml up -d --build
+      run: (set -a && . ./env.test && set +a && docker compose -f docker-compose.test.yml up -d)
     - name: Run tests with Coverage
-      run: pytest --cov=. --cov-report=xml
+      run: (set -a && . ./env.test && set +a && python -m pytest -s -v --cov=. --cov-report=xml)
     - name: Stop services provided by Docker Compose
-      run: docker compose -f docker-compose.testing.yml down
+      run: (set -a && . ./env.test && set +a && docker compose -f docker-compose.test.yml down)
```

Comment:
@jwodder is there a more convenient way to inherit changes in environment in one step across all steps?   pretty much ./env.test has variables which we need to export/use in those steps

Reply:
@yarikoptic If all the lines in `env.test` are of the form `varname=value`, you can append it to [the file in `$GITHUB_ENV`](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable).

Reply:
One reason I want to do it this way is that it is more inline with the instruction for setting up testing at the [README](https://github.com/datalad/datalad-registry?tab=readme-ov-file#to-run-tests).  The only difference between this line and the one in the README instruction is the use of `docker compose` instead of `podman-compose`.

This arrangement can be better at helping us to pick up some inconsistencies between our code and documentation.

Reply:
I can see how it could be beneficial, but my allergy to duplication and clutter which then makes it harder to find defects almost outweighs a possible benefit from a distant harmonization with the README, since in effect it would be achieving the same.

If you were to test "instructions in README", here is an example how I do it https://github.com/ReproNim/containers/blob/master/.github/workflows/tests.yaml#L30 -- I really take and run it.  Duplicating (duplication is evil) instruction just opens the way for them to diverge eventually.

So overall  - even if motivation is sound, long term IMHO it is not beneficial and might be confusing for someone looking at it ("why they didn't just `cat env.test >>  "$GITHUB_ENV"`?) and  it is **already** different from README (docker vs podman).